dotnet/src/webdriver/WebDriver.cs

```diff
@@ -854,6 +851,15 @@ private static void UnpackAndThrowOnError(Response errorResponse, string command
                         case WebDriverResult.InsecureCertificate:
                             throw new InsecureCertificateException(errorMessage);
 
+                        case WebDriverResult.UnknownError:
+                            throw new UnknownErrorException(errorMessage);
+
+                        case WebDriverResult.UnknownMethod:
+                            throw new UnknownMethodException(errorMessage);
+
+                        case WebDriverResult.UnsupportedOperation:
+                            throw new UnsupportedOperationException(errorMessage);
+
                         default:
                             throw new InvalidOperationException(string.Format(CultureInfo.InvariantCulture, "{0} ({1})", errorMessage, errorResponse.Status));
```

Comment:
Is this exception reachable? If so, should it be `WebDriverException` instead of invalid operation exception?

Reply:
It might be, we cannot always map server-side error with whatever we can handle. Usually it is related to when we send some command and get result. From my perspective it is OK to throw some default exception, even if it is out of spec. Some providers, which are spec compatible, might not follow spec.